### ISSUE TITLE
Support creating client with Streams

### ISSUE THREAD

**Issue by empirefox** [grpc/grpc-dart#115]
It is easier to invoke golang code from flutter code. Due to less ios/android code and class-struct mapping, coding errors will be reduced.

**Comment by gedw99** [comment]
Not sure what you mean. Can you be more detailed ?

**Comment by empirefox** [comment]
When calling golang api from flutter, currently we must write the ios/android version api for every golang api. If it is possible to create grpc client using stream and sink, we can make dart-golang calls easier, because we only need to establish a virtual connection(stream and sink) to golang code through the platform channel. The android/ios code do not need to be changed when a golang api changes. I'm sorry for my poor English.

**Comment by gedw99** [comment]
now i understand what you mean.
You want to run golang inside flutter ?

**Comment by empirefox** [comment]
Yes. And i want to make it more easier to maintain for me. I am not familiar for android/ios, so I want make the the platform code to be decoupled from the business.

**Comment by Mohammed90** [comment]
What you're asking for is not doable yet due to limitation in Flutter, not grpc-Dart. This is evident by the discussion in Flutter in the links below:

https://github.com/flutter/flutter/issues/2396

https://github.com/flutter/flutter/issues/7053

**Comment by empirefox** [comment]
Yes, flutter call C/C++ directly is impossible. But using [platform channel](https://flutter.io/platform-channels/) to only pass binary data is feasible.

**Comment by empirefox** [comment]
I created a flutter [plugin](https://github.com/empirefox/flutter_dial_go), which can create a pair of (StreamSink<List<int>>, Stream<List<int>>). From flutter it can establish a http2 connection:

```dart
    // golang: ln,err := forgo.Listen(9999)
    _conn = await Conn.dial(9999);
    _conn.done.then((_) {
      _conn.close();
    });
    _transport = ClientTransportConnection.viaStreams(_conn.receiveStream, _conn);
```

So i need a way to create grpc client from that `_transport` or `_conn`.

**Comment by gedw99** [comment]
https://github.com/flutter/flutter/issues/7053 has now been solved btw !!!
